apps/dotcms-ui/src/app/api/services/add-to-menu/add-to-menu.service.ts

```diff
@@ -0,0 +1,94 @@
+import { pluck, catchError, take, map } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { CoreWebService } from '@dotcms/dotcms-js';
+import { DotHttpErrorManagerService } from '@services/dot-http-error-manager/dot-http-error-manager.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+const addToMenuUrl = `v1/portlet`;
+
+export interface DotCreateCustomTool {
+    contentTypes: string;
+    dataViewMode: string;
+    portletId: string;
+    portletName: string;
+}
+
+/**
+ * Provides methods to create and assign custom tools portlet to layout menu.
+ * @export
+ * @class DotAddToMenuService
+ */
+@Injectable()
+export class DotAddToMenuService {
+    constructor(
+        private coreWebService: CoreWebService,
+        private httpErrorManagerService: DotHttpErrorManagerService
+    ) {}
+
+    /**
+     * Cleans the portletId string from special chars and replaces then with a dash
+     * @param {string} name
+     * @returns string
+     * @memberof DotAddToMenuService
+     */
+    cleanUpPorletId(name: string): string {
+        const cleanSpecialChars = name.replace(/[^0-9a-z-]/gi, '-');
+        const cleanPortlet = cleanSpecialChars.replace(/(?:\s*-\s*)+|\s{2,}/gm, '-');
+        return cleanPortlet;
+    }
+
+    /**
+     * Creates a Custom tool portlet
+     * @param {DotCreateCustomTool} params
+     * @returns Observable<string>
+     * @memberof DotAddToMenuService
+     */
+    createCustomTool(params: DotCreateCustomTool): Observable<string> {
```

Comment:
What is this string? is there any values that we can have to an `enum` or something like that?

Reply:
based on the `params` it returns the name of the created portlet in the BE